pages/curriculum/[lesson].tsx

```diff
@@ -15,10 +16,20 @@ import _ from 'lodash'
 
 const Challenges: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
   const { lessons, session, alerts } = queryData
-  const userSubmissions: UserSubmission[] = _.get(session, 'submissions', [])
-  const lessonStatus: LessonStatus[] = _.get(session, 'lessonStatus', [])
   const router = useRouter()
   const currentlessonId = router.query.lesson as string
+  if (!session || !lessons || !alerts) {
+    return (
+      <Error title="Internal server error" message="Bad data" src="/500.png" />
+    )
+  }
+  if (
+    !lessons.find((lesson: Lesson) => lesson.id.toString() === currentlessonId)
```

Comment:
`lesson.id.toString()` is the `toString` needed? In our typedefs, `id` is defined as a string already, so id should already be a string
https://github.com/garageScript/c0d3-app/blob/master/graphql/typeDefs.ts#L130

Reply:
It's a bug, thanks for noticing.
 Lesson was typed twice, first as gql query in your link and then as `Lesson` type in https://github.com/garageScript/c0d3-app/blob/master/%40types/lesson.d.ts. Since app.d.ts [uses](https://github.com/garageScript/c0d3-app/blob/master/%40types/app.d.ts) latter Lesson type with number id I need to convert it to string first otherwise I get typerror. Meanwhile actual query comes with string ids. 
I changed Lesson id type to string and removed all `toString` conversions.

Reply:
Thanks for fixing the bug!
Blah, I think it is confusing to have `ids` as strings instead of numbers.. but all the ids in our typedefs file are defined as strings, even though they are an integer in the database. I find it confusing -- I think that our typedefs should match the data types of what is defined in our database for increased clarity.